src/main/java/com/project/bookapp/controller/CartController.java

```diff
@@ -0,0 +1,67 @@
+package com.project.bookapp.controller;
+
+import com.project.bookapp.dto.cart.ShoppingCartResponseDto;
+import com.project.bookapp.dto.item.CartItemRequestDto;
+import com.project.bookapp.dto.item.CartItemResponseDto;
+import com.project.bookapp.service.CartItemService;
+import com.project.bookapp.service.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Cart management", description = "Endpoints for cart")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/cart")
+public class CartController {
+
+    private final ShoppingCartService shoppingCartService;
+    private final CartItemService cartItemService;
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping("/{id}")
+    @Operation(summary = "Get a user's shopping cart",
+            description = "Get a user's shopping cart")
+    public ShoppingCartResponseDto getById(@PathVariable Long id) {
```

Comment:
check task description
`GET: /api/cart (Retrieve user's shopping cart)`

```suggestion
    public ShoppingCartResponseDto getShoppingCart(Authentication authentication) {
```